components/tel/Core/Src/xbee.c

```diff
@@ -0,0 +1,87 @@
+/**
+ *  @file xbee.c
+ *  @brief Defines functions related to xBee module, including setting up xBee packet overhead, and handling return frames
+ *
+ *  @date 2024/06/16
+ *  @author Evan Owens
+ */
+
+#include "xbee.h"
+#include <math.h>
+
+/**
+ * @brief Adds up all the bytes after length and before checksum in an api packet
+ *
+ * @param api packet: api packet array
+ * @return uint16_t total_bytes in api packet before checksum and after length fields
+*/
+
+uint16_t XBEE_sum_bytes (uint8_t api_packet[], uint16_t packet_length)
+{
+  uint16_t total_bytes = 0;
+  for (int i = 3; i < packet_length; i++){
```

Comment:
Add a symbolic constant here: `for (int i = 3; i < packet_length; i++){` for the number 3. Also this implies that the checksum is at the first 4 bytes of the API packet but I see in the overhead set up function that your first 4 bytes are delimeter, frame type, frame id, and bit address low.  I also thought that the checksum came at the end of the message (correct me on this)

Reply:
to calculate the checksum, you add up all the bytes from after the length so position 3 all the way to the end of the packet (before the checksum). So by doing what i do above, I am just ignoring the start delimiter, msb, and lsb. xctu or the api bom provides a more clear breakdown as my function does not necessarily add all the fields in order.

Reply:
Makes sense. Thanks for clarification.

Reply:
@EvanO12 I would add this clarification in the function brief so anyone looking at the function knows exactly how the checksum is calculated.